src/Traits/HasRoles.php

```diff
@@ -96,14 +96,13 @@ public function scopeRole(Builder $query, $roles, $guard = null): Builder
     }
 
     /**
-     * Assign the given role to the model.
+     * Returns roles ids as array keys
      *
-     * @param  array|string|int|\Spatie\Permission\Contracts\Role|\Illuminate\Support\Collection  ...$roles
-     * @return $this
+     * @param  array|string|int|\Spatie\Permission\Contracts\Role|\Illuminate\Support\Collection  $roles
      */
-    public function assignRole(...$roles)
+    private function collectRoles(...$roles): array
```

Comment:
in `HasPermissions` the "collect" method was `public`.
I agree with keeping it `private` whenever possible. Should the other be made private too?

Reply:
if it would be the best